src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Class containing the main Duke application.
+ * Initialises the application.
+ */
+public class Duke {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Storage st = new Storage("./data/dukedata.txt");
+        TaskList tl = new TaskList(st.setFile());
+        Ui ui = new Ui();
+
+        ui.printIntro();
+
+        while (true) {
+            String curr = sc.nextLine();
+            String[] msgWords = curr.split("\\b");
+            for (int i = 0; i < msgWords.length; i++) {
+                if (msgWords[i].equalsIgnoreCase("bye")) {
+                    sc.close();
+                    ui.goodByeMessage();
+                    return;
+                } else if (msgWords[i].equalsIgnoreCase("list")) {
+                    ui.printList(tl);
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("mark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
+                        ui.printOutOfBoundsMsg();
+                        break;
+                    }
+                    tl.mark(index);
+                    ui.printMarkedMsg(tl.get(index));
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("unmark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
+                        ui.printOutOfBoundsMsg();
+                        break;
+                    }
+                    tl.unMark(index);
+                    ui.printUnmarkedMsg(tl.get(index));
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("delete")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    Task removed = tl.get(index);
+                    tl.remove(index);
+                    ui.printDeleteMsg(removed.toString(), tl.size());
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("todo")) {
+                    if (curr.length() < 6) {
+                        ui.printNoTaskInputMsg();
+                        break;
+                    }
+                    String item = curr.substring(4);
+                    Todo newTodo = new Todo(item);
+                    tl.add(newTodo);
+                    ui.printTaskAddedMsg(newTodo, tl.size());
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("deadline")) {
+                    if (curr.length() < 10 || !curr.contains("/by")) {
```

Comment:
I think you can group the methods in like tasks or something. 
Eg.  In parser: task.addDeadline();
In Task: void addDeadline(){}